storage/database/src/main/java/de/danoeh/antennapod/storage/database/ReleaseScheduleGuesser.java

```diff
@@ -122,13 +126,56 @@ private static Stats getStats(List<Date> releaseDates) {
                 daysOfWeek, daysOfMonth, mostOftenDayOfWeek, mostOftenDayOfMonth);
     }
 
+    private static List<Integer> getLargeDays(ReleaseScheduleGuesser.Stats stats, int maxDaysOff) {
+        List<Integer> largeDays = new ArrayList<>();
+        for (int i = Calendar.SUNDAY; i <= Calendar.SATURDAY; i++) {
+            if (stats.daysOfWeek[i] > maxDaysOff) {
+                largeDays.add(i);
+            }
+        }
+        return largeDays;
+    }
+
+    private static Date getNormalizedDate(Date date) {
+        GregorianCalendar current = new GregorianCalendar();
+        current.setTime(date);
+        current.set(Calendar.HOUR_OF_DAY, 0);
+        current.set(Calendar.MINUTE, 0);
+        current.set(Calendar.SECOND, 0);
+        current.set(Calendar.MILLISECOND, 0);
+        return current.getTime();
+    }
+
     public static Guess performGuess(List<Date> releaseDates) {
         if (releaseDates.size() <= 1) {
-            return new Guess(Schedule.UNKNOWN, null, null);
-        } else if (releaseDates.size() > MAX_DATA_POINTS) {
-            releaseDates = releaseDates.subList(releaseDates.size() - MAX_DATA_POINTS, releaseDates.size());
+            return new Guess(Schedule.UNKNOWN, null, null, false);
+        } else if (releaseDates.size() > MAX_UNIQUE_DATES) {
+            Set<Date> uniqueDates = new HashSet<>();
+            int i = releaseDates.size();
+            int uniqueDateCounter = MAX_UNIQUE_DATES;
+            while (i > 0 && uniqueDateCounter > 0) {
```

Comment:
Wouldn't `uniqueDates.size()` be a bit easier to understand than having an additional `uniqueDateCounter`?